src/hayhooks/server/utils/deploy_utils.py

```diff
@@ -370,6 +301,58 @@ def add_pipeline_api_route(app: FastAPI, pipeline_name: str, pipeline_wrapper: B
     app.setup()
 
 
+def add_pipeline_yaml_api_route(app: FastAPI, pipeline_name: str) -> None:
+    """
+    Create or replace the YAML pipeline run endpoint at /{pipeline_name}/run.
+
+    Builds the flat request/response models from declared YAML inputs/outputs and wires a handler that
+    maps the flat body into the nested structure required by Haystack Pipeline.run.
+    """
+    pipeline_instance = registry.get(pipeline_name)
+    if pipeline_instance is None:
+        msg = f"Pipeline '{pipeline_name}' not found"
+        raise PipelineNotFoundError(msg)
+
+    # Ensure the registered object is a Haystack Pipeline, not a wrapper
+    if not isinstance(pipeline_instance, AsyncPipeline):
+        msg = f"Pipeline '{pipeline_name}' is not a Haystack AsyncPipeline instance"
+        raise PipelineYamlError(msg)
+
+    pipeline: AsyncPipeline = pipeline_instance
+    metadata = registry.get_metadata(pipeline_name) or {}
+
+    PipelineRunRequest = metadata.get("request_model")
+    PipelineRunResponse = metadata.get("response_model")
+
+    if PipelineRunRequest is None or PipelineRunResponse is None:
+        msg = f"Missing request/response models for YAML pipeline '{pipeline_name}'"
+        raise PipelineYamlError(msg)
+
+    @handle_pipeline_exceptions()
+    async def pipeline_run(run_req: PipelineRunRequest) -> PipelineRunResponse:  # type:ignore[valid-type]
+        result = await pipeline.run_async(data=run_req.model_dump())  # type: ignore[attr-defined]
```

Comment:
nit: I would add short comments explaining why we need `type:ignore`

Reply:
yes, reason is the same stated in `create_run_endpoint_handler` docstring. We may add a reference to it on the top of this method, or simply cut/paste